libraries/networking/src/NetworkAccessManager.cpp

```diff
@@ -11,18 +11,42 @@
 
 #include "NetworkAccessManager.h"
 
-#include <QThreadStorage>
-
 #include "AtpReply.h"
+#include "ResourceCache.h"
+
+#include <QStandardPaths>
+#include <QThreadStorage>
 #include <QtNetwork/QNetworkProxy>
+#include <QtNetwork/QNetworkDiskCache>
+
+#include <shared/GlobalAppProperties.h>
 
 QThreadStorage<QNetworkAccessManager*> networkAccessManagers;
 
 QNetworkAccessManager& NetworkAccessManager::getInstance() {
     if (!networkAccessManagers.hasLocalData()) {
-        networkAccessManagers.setLocalData(new QNetworkAccessManager());
+        auto networkAccessManager = new QNetworkAccessManager();
+        // Setup disk cache if not already
+        if (!networkAccessManager->cache()) {
+            QString cacheDir = qApp->property(hifi::properties::APP_LOCAL_DATA_PATH).toString();
+            if (cacheDir.isEmpty()) {
+#ifdef Q_OS_ANDROID
+                QString cachePath = QStandardPaths::writableLocation(QStandardPaths::CacheLocation);
+#else
+                QString cachePath = QStandardPaths::writableLocation(QStandardPaths::DataLocation);
+#endif
+                cacheDir = !cachePath.isEmpty() ? cachePath : "interfaceCache";
+            }
+            QNetworkDiskCache* cache = new QNetworkDiskCache();
+            cache->setMaximumCacheSize(MAXIMUM_CACHE_SIZE);
+            cache->setCacheDirectory(cacheDir);
+            networkAccessManager->setCache(cache);
```

Comment:
according to the docs: https://doc.qt.io/qt-5/qnetworkaccessmanager.html#setCache

"QNetworkAccessManager by default does not have a set cache."

so...I think we were forcing PreferCache above, but then...not actually setting up a cache?  hopefully this new code is more correct and more performant